packages/medusa/src/api-v2/admin/product-types/route.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "../../../types/routing"
+
+import {
+  ContainerRegistrationKeys,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
+import {
+  AdminGetProductTypesParams,
+  AdminPostProductTypesReq,
+} from "./validators"
+import { createProductTypesWorkflow } from "@medusajs/core-flows"
+
+export const GET = async (
+  req: AuthenticatedMedusaRequest<AdminGetProductTypesParams>,
+  res: MedusaResponse
+) => {
+  const remoteQuery = req.scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+  const queryObject = remoteQueryObjectFromString({
+    entryPoint: "product",
+    variables: {
+      filters: req.filterableFields,
+      ...req.remoteQueryConfig.pagination,
+    },
+    fields: req.remoteQueryConfig.fields,
+  })
+
+  const { rows: product_types, metadata } = await remoteQuery(queryObject)
+
+  res.json({
+    product_types: product_types,
+    count: metadata.count,
+    offset: metadata.skip,
+    limit: metadata.take,
+  })
+}
+
+export const POST = async (
+  req: AuthenticatedMedusaRequest<AdminPostProductTypesReq>,
+  res: MedusaResponse
+) => {
+  const input = [req.validatedBody]
+
+  const { result, errors } = await createProductTypesWorkflow(req.scope).run({
+    input: { product_types: input },
+    throwOnError: false,
+  })
+
+  if (Array.isArray(errors) && errors[0]) {
+    throw errors[0].error
+  }
+
+  res.status(200).json({ product_type: result[0] })
```

Comment:
@olivermrbl do you mean to refetch the data? there is no need in this case I believe, as the `type` model doesn't have any relations and it only has 2 fields to modify, but I am happy to add it if you think we should

Reply:
It would be for the sake of consistency. If someone links to a `product_type_id`, they won't be able to query the relations on create, as in most other endpoints. This is an edge case, so fine to keep it out. I just like the idea of having the same query API in all APIs, so we won't ever be in a situation where someone asks for it and we need to add it back in.